### PR TITLE
Link to form docs on platform website

### LINKED ISSUE #755: Point users to the Forms System Documentation from Components > Form and Patterns > Forms
We need to direct users to the Forms System Documentation: 
https://depo-platform-documentation.scrollhelp.site/developer-docs/VA-Forms-System-Overview.2085355587.html

* [x] Add link to documentation on Components > Form
* [x] Add link to documentation on Patterns > Forms

Use the Featured Content component for this with some text explaining the current state of the Forms System.

### PR DESCRIPTION
**Pull request by bkjohnson**
## Description

Closes #755 

Following the [usage guidance](https://design.va.gov/components/featured-content#usage), I'm adding a featured content component with an `<h3>` heading underneath the first `<h2>` on the page. In order to do this I also changed the additional `<h1>` heading on the form patterns page to be an `<h2>`.

### Form components page

Highlighting the featured content on the components/forms page

### Form patterns page

Highlighting the featured content on the patterns/forms page

### CODE REVIEW

File: src/_about/for-developers.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ slug: developers
 
 ## Using the Design System
 
-**If you are working in the vets-website repository**, you can skip straight to the [developer documentation](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/). Otherwise, proceed below.
+**If you are working in the vets-website repository**, you can skip straight to the [developer documentation](https://depo-platform-documentation.scrollhelp.site/developer-docs/Frontend-developer-documentation.687931428.html). Otherwise, proceed below.
```
Comment (bkjohnson) [inline src/_about/for-developers.md:30]: Platform site for frontend developers.

File: src/_patterns/forms/index.md [hunk 2]
```diff
@@ -62,7 +68,7 @@ First things first. Forms on VA.gov typically consist of:
 
 For numbers 3-4 mentioned above, users are given guidance on where they are in the form using a segmented progress bar.  Each segment of the bar represents a “chapter” in the form.  Note that there can be multiple “pages” within each chapter, so the bar does not always increment when the user clicks between pages.
 
-For a bit of context on the technical side, [read up on the form system](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/forms/) and/or [watch the quick demo Chris V. gave to the VSA design team](https://zoom.us/rec/share/x5FpP7XC1DJOYZ3d933GUagdMInqeaa82ilM-KcLzUxo_-q1CWuJcUJVwzaPXTp_) in Feb. 2020 (pw: vsadesign) 
+For a bit of context on the technical side, [read up on the form system](https://depo-platform-documentation.scrollhelp.site/developer-docs/VA-Forms-Library-Overview.2085355587.html) and/or [watch the quick demo Chris V. gave to the VSA design team](https://zoom.us/rec/share/x5FpP7XC1DJOYZ3d933GUagdMInqeaa82ilM-KcLzUxo_-q1CWuJcUJVwzaPXTp_) in Feb. 2020 (pw: vsadesign)
```
Comment (bkjohnson) [inline src/_patterns/forms/index.md:71]: Updating this link to point to the new platform website. Do we still want this now that we have the link in the featured content?

File: src/_patterns/forms/index.md [hunk 3]
```diff
@@ -198,10 +204,9 @@ This document will not attempt to rewrite those documents, however, we will prov
 * Make sure you read up on and understand form builder:
    *  Fill out a few forms on staging.va.gov to get a feel for the capabilities of the forms system.
    *  [Watch the zoom video of a quick demo Chris Valarida gave to the design team](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/vsa/design/va-forms-informal-for-designers.mp4) in Feb. 2020. He goes over an example JSON object used by react-jsonschema-form (RJSF), and toggles between it and the rendered UI, so developers may find it useful.
-   *  Before coding, start with the VA<span/>.gov Forms System (VAFS) documentation on the [VA<span/>.gov Client Application Documentation site](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/): **https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/forms**
```
Comment (bkjohnson) [inline src/_patterns/forms/index.md:201]: I removed this because of the featured content link :shrug:

File: src/_patterns/forms/index.md [hunk 4]
```diff
@@ -198,10 +204,9 @@ This document will not attempt to rewrite those documents, however, we will prov
...
    *  If you have determined that you need to implement custom behavior or appearance that the existing components do not support out-of-the-box, FIRST confirm with your design team that the feature is needed and cannot be implemented an alternate way that fits within the limitations of the existing forms framework. The va<span/>.gov Forms System (VAFS) is built on top of the VA's fork of react-jsonschema-form (RJSF), so if customization is required (or if you need a deeper understanding that goes beyond the tutorial), then you will need to dig one level deeper:
-      *  Read the overview [Creating custom fields and widgets](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/forms/creating-custom-fields-and-widgets), THEN
+      *  Read the overview [Creating custom fields and widgets](https://depo-platform-documentation.scrollhelp.site/developer-docs/VA-Forms-Library---How-to-create-custom-fields-and-widgets.2115567784.html), THEN
```
Comment (bkjohnson) [inline src/_patterns/forms/index.md:209]: Updating an older link to point to the page on the platform documentation site.

File: src/index.html [hunk 5]
```diff
@@ -110,7 +110,7 @@ <h3 class="vads-u-margin--0 vads-u-font-family--sans">Developer.va.gov</h3>
         </a>
       </div>
       <div class="vads-l-col--12 medium-screen:vads-l-col--6 vads-u-padding--2">
-        <a href="https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/" class="site-resource-link">
+        <a href="https://depo-platform-documentation.scrollhelp.site/developer-docs/" class="site-resource-link">
```
Comment (bkjohnson) [inline src/index.html:113]: Another update not specific to the forms library documentation.

**Review by bkjohnson — COMMENTED** [review]
Calling out some changes that aren't related to the link in the featured content.

**Review by k80bowman — CHANGES_REQUESTED** [review]
Could you use `site.forms_system_link` for all references? I see you use for some but others are hard-coded.

**Comment by bkjohnson** [comment]
Those other links are for different pages. The `site.forms_system_link` is for `/developer-docs/VA-Forms-System-Overview.2085355587.html`:

https://github.com/department-of-veterans-affairs/vets-design-system-documentation/blob/d5848c24c9977fe758c38c43889af2196209aa69/_config.yml#L12

And we also have:

- `/developer-docs/`
- `/developer-docs/Frontend-developer-documentation.687931428.html`
- `/developer-docs/VA-Forms-Library---How-to-create-custom-fields-and-widgets.2115567784.html`

**Review by k80bowman — APPROVED** [review]
Ah ok. Sorry, didn't realize

**Pull request merged**